include/pybind11/detail/internals.h

```diff
@@ -311,12 +311,23 @@ PYBIND11_NOINLINE inline internals &get_internals() {
     return **internals_pp;
 }
 
+
+struct module_internals {
+  type_map<type_info *> registered_local_types_cpp;
+  std::forward_list<void (*) (std::exception_ptr)> registered_local_exception_translators;
+};
+
+inline module_internals &get_module_internals() {
+  static module_internals locals;
+  return locals;
+}
+
 /// Works like `internals.registered_types_cpp`, but for module-local registered types:
 inline type_map<type_info *> &registered_local_types_cpp() {
-    static type_map<type_info *> locals{};
-    return locals;
+    return get_module_internals().registered_local_types_cpp;
```

Comment:
Not sure how other parts of pybind11 do it, but you do have a duplicate use of `registered_local_types_cpp` here (field as well as the function name). Can you check what conventions other parts of `internals.h` follow?

Reply:
Well, I overcame my laziness and looked up myself. Can we go for `get_registered_local_types_cpp` as function name, for example? I don't think this should break ABI things.

Reply:
would you prefer me to rename this function or just replace usages of it with `get_module_internals().registered_local_types_cpp;` so it works more like get_internals?

Reply:
or would that break ABI? I don't think this function should be ever used in user code

Reply:
Matching `get_internals()` closer might be nice, indeed, yes, if you ask me :-)
And no, this shouldn't break user code (since it's in the `detail` namespace), and ... yeah, no, the whole point is that this isn't shared between modules, so there's no shared ABI :-)

Reply:
I have updated this to be called `get_local_internals()`, deleted `get_registered_local_types_cpp()` completely, and changed the usage to be more like `get_internals()`. Do we want to leave the double `local` in the names or match the member names to the `internals` member names?
currently:
```
get_local_internals().registered_local_types_cpp;
```
proposed change to remove the local in the member name
```
get_local_internals().registered_types_cpp;
```